### PR TITLE
feat: Build Typescript Pagination Function 

### PR DESCRIPTION
**Pull request by claudiachua**
Internal JIRA Link: [DI-1401](https://issues.corp.twilio.com/browse/DI-1401)
Related PR:  https://github.com/twilio/twilio-node/pull/797

### CODE REVIEW

File: src/main/resources/twilio-node/operation.mustache [hunk 1]
```diff
@@ -49,23 +53,61 @@
...
+        };
+        let operationPromise = new Promise((resolve, reject) => {
+            params.done = function (error) {
+                if (typeof error === 'undefined') {
+                    resolve(allResources);
+                } else {
+                    reject(error);
+                }
+            };
+        });
```
Comment (childish-sambino) [inline src/main/resources/twilio-node/operation.mustache:98]: Can this logic be moved to a helper like the `each` call?

Comment (claudiachua) [reply src/main/resources/twilio-node/operation.mustache:98]: Will see if I can do it with Generics as the return type is specific unlike each

**Review by beebzz — COMMENTED** [review]
Tests are probably failing b/c the example code uses the `version.js` file from the RC branch (which doesn't include `setPromiseCallback()`), and not your feature branch

File: src/main/resources/twilio-node/api-single.mustache [hunk 2]
```diff
@@ -93,14 +97,51 @@ export class {{resourceName}}Impl implements {{resourceName}} {
     return inspect(this.toJSON(), options);
   }
 }
-{{/parentResourceName}}
 
+export interface {{name}}Solution {
+{{#resourcePathParams}}
+    {{paramName}}?: {{{dataType}}};
```
[the remark this answers is not in the document]
Comment (claudiachua) [reply src/main/resources/twilio-node/api-single.mustache:103]: There were a couple instances that Solution encapsules different property. I figured making it optional would be the best way forward. 

[ConferenceInstance._solution](https://github.com/twilio/twilio-node/blob/main/lib/rest/api/v2010/account/conference.js#L511) required both accountSid and sid 
[ConferenceListInstance._solution](https://github.com/twilio/twilio-node/blob/main/lib/rest/api/v2010/account/conference.js#L55) only required accountSid

**Comment by claudiachua** [comment]
run pr validation

**Review by childish-sambino — APPROVED** [review]

**Pull request merged**